src/EnergyPlus/LowTempRadiantSystem.cc

```diff
@@ -4699,6 +4734,40 @@ namespace LowTempRadiantSystem {
 
     }
 
+    void ConstantFlowRadiantSystemData::calculateRunningMeanAverageTemperature()
+    {
+        if (DataGlobals::DayOfSim == 1 && DataGlobals::WarmupFlag) {
+            // there is no "history" here--assume everything that came before was the same (this applies to design days also--weather is always the same
+            this->todayAverageOutdoorDryBulbTemperature = this->calculateCurrentDailyAverageODB();
+            this->yesterdayAverageOutdoorDryBulbTemperature = this->todayAverageOutdoorDryBulbTemperature;
+            this->todayRunningMeanOutdoorDryBulbTemperature = this->todayAverageOutdoorDryBulbTemperature;
+            this->yesterdayRunningMeanOutdoorDryBulbTemperature = this->todayAverageOutdoorDryBulbTemperature;
+        } else if (!DataGlobals::WarmupFlag && DataGlobals::NumOfDayInEnvrn > 1) {
+            // This is an environment with more than one day (non-design day) so...
+            // First update yesterday's information using what was previously calculated for "today"
+            this->yesterdayAverageOutdoorDryBulbTemperature = this->todayAverageOutdoorDryBulbTemperature;
+            this->yesterdayRunningMeanOutdoorDryBulbTemperature = this->todayRunningMeanOutdoorDryBulbTemperature;
+            // Now update the running mean and average outdoor air temperatures
+            //NOTES: ALPHA COULD BE DIFFERENT FOR EACH SYSTEM BECAUSE IT WILL BE A FACTOR OF EACH SYSTEM--NEED TO CHANGE LOCATION OF SOME OF THESE VARIABLES
+            //       AS A RESULT (RUNNING MEAN TEMPERATURES).  alpha WILL NEED TO BE REPLACED BY APPROPRIATE TERM IN CONSTANT FLOW DATA AND STILL NEED TO READ
+            //       THIS VALUE IN IN THE GET ROUTINE (AND MODIFY IDD, INPUT FILES, TRANSITION, ETC.)
+            this->todayRunningMeanOutdoorDryBulbTemperature = (1.0 - this->runningMeanOutdoorAirTemperatureWeightingFactor) * this->yesterdayAverageOutdoorDryBulbTemperature
+                                                              + this->runningMeanOutdoorAirTemperatureWeightingFactor * this->yesterdayRunningMeanOutdoorDryBulbTemperature;
+            this->todayAverageOutdoorDryBulbTemperature = this->calculateCurrentDailyAverageODB();
+        }
+    }
+
+    Real64 ConstantFlowRadiantSystemData::calculateCurrentDailyAverageODB()
+    {
+        int const firstTimeStepIndex = 1;
+        Real64 sum = 0.0;
+        for (int hourNumber = 1; hourNumber <= DataGlobals::HoursInDay; ++hourNumber) {
+            sum += WeatherManager::TodayOutDryBulbTemp(firstTimeStepIndex,hourNumber);
```

Comment:
So, the average is only using values for the first timestep in each hour? If the epw file is hourly, these won't match the epw values (those appear at the last timestep of each hour). If the epw file is sub-hourly, is this ok?
Also, so this looks ahead at the full day? Isn't that beyond what a real control system could do, or do they use predicted weather to control the actual system? I must be missing something here.

From the I/O Ref description, it seems like this should be averaging yesterday's temperatures, no Today's.

Reply:
If you check where the temperature is used, it's actually using yesterday's temperature.  I'm calculating "today's" after moving today's temperature to the yesterday value at the start of each day.  So, yes, I'm calculating today's value but it never gets used until the start of the next day when the yesterday value is then set.  Does that make sense now?

As for the averaging, my bad--I can switch it to the last time step in the hour.  I'll look at whether or not there is something that indicates whether the weather values are hourly or sub hourly.  I believe that most standard (available from the E+ web site) are hourly values.  As you can see, my understanding of the weather stuff is not complete, but I'll dig a bit more and make whatever change I can make.  Or in reality, I guess I could just average the time step values since they have been "averaged" by the program anyway?

Changes should show up in the next commit.

Reply:
OK. A comment here to explain would be helpful. And regarding the first/last timestep of the hour - you need to double-check my understanding. It's can be a bit messy. And then there's the case of sub-hourly weather files. So, in order to keep this simple, why not just average at each timestep instead of hourly?

Reply:
Yes, I'll add a comment as to why I'm calculating today and how it's used.

As for the averaging, yes, I can and will do it on the time step.  When I started the work, I thought it was loading it into the first time step.  Now, I'm not so sure and it doesn't really matter since we should allow for sub-hourly weather either way.  So, I'll implement a time step averaging scheme and move forward with that.